geonode/upload/tests/end2end/test_end2end.py

```diff
@@ -488,3 +494,61 @@ def test_import_wms(self):
             "ptype": "gxp_wmscsource",
         }
         self._assertimport(payload, initial_name, skip_geoserver=True, assert_payload=assert_payload)
+
+    @mock.patch.dict(os.environ, {"GEONODE_GEODATABASE": "test_geonode_data", "ASYNC_SIGNALS": "False"})
+    @override_settings(GEODATABASE_URL=f"{geourl.split('/geonode_data')[0]}/test_geonode_data", ASYNC_SIGNALS=False)
+    def test_import_wms_harvestable_resource_should_be_created(self):
+        """
+        The WMS import should generate the harvestable resource if the remote service URL
+        match with an exists harvester, so next time it will be managed by the harvester itself
+        """
+
+        # creating remote service
+        # for the test we have to pass via the proxy
+        harvester = Harvester.objects.create(
+            remote_url="http://localhost:8000/proxy/",
+            name="Test",
+            default_owner=self.user,
+            harvester_type="geonode.harvesting.harvesters.wms.OgcWmsHarvester",
+        )
+        _, wms = WebMapService(
+            f"{os.getenv('GEOSERVER_LOCATION')}ows?service=WMS&version=1.3.0&request=GetCapabilities"
+        )
+        resource_to_take = next(iter(wms.contents))
+        res = wms[next(iter(wms.contents))]
+        import urllib.parse
+
+        url = urllib.parse.quote(
+            f"{os.getenv('GEOSERVER_LOCATION')}ows?service=WMS&version=1.3.0&request=GetCapabilities", safe=""
+        )
+        payload = {
+            "url": f"http://localhost:8000/proxy/?url={url}",
```

Comment:
NOTE: the django serializer raises error if the url povided of the remote service is `geoserver:8080` since is not a valid URL, in order to make the test pass, we need to pass via the proxy since `localhost:8000` is a valid url